### PR TITLE
[doc] Improve Pulsar Security Encryption

### PR DESCRIPTION
**Pull request by Monica-zy**
Motivation
Improve the language and the overall descriptive style of the Pulsar Security document (security-encryption section): http://pulsar.apache.org/docs/en/next/security-encryption/

Modifications
Adjust the tone, personal pronouns, voice also some typo errors of some sentences in the document.

### CODE REVIEW

**Comment by Monica-zy** [comment]
@Jennifer88huang Please review this, thank you~

File: site2/docs/security-encryption.md [hunk 1]
```diff
@@ -4,19 +4,19 @@ title: Pulsar Encryption
 sidebar_label: End-to-End Encryption
 ---
 
-Pulsar encryption allows applications to encrypt messages at the producer and decrypt at the consumer. Encryption is performed using the public/private key pair configured by the application. Encrypted messages can only be decrypted by consumers with a valid key.
+Pulsar encryption allows applications to encrypt messages at the producer side and decrypt messages at the consumer side. You can use the public and private key pair that the application configures to perform encryption. Only the consumers with a valid key can decrypt the encrypted messages.
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: do not use "allow".
Check styles in IBM style guides.

File: site2/docs/security-encryption.md [hunk 2]
```diff
@@ -4,19 +4,19 @@ title: Pulsar Encryption
...
+The application configures the producer with the public key. You can use this key to encrypt the AES data key. The encrypted data key is sent as part of message header. Only entities with the private key (in this case the consumer) are able to decrypt the data key which is used to decrypt the message.
 
-A message can be encrypted with more than one key.  Any one of the keys used for encrypting the message is sufficient to decrypt the message
+You can encrypt a message with more than one key. Any one of the keys used for encrypting the message is sufficient to decrypt the message.
 
-Pulsar does not store the encryption key anywhere in the pulsar service. If you lose/delete the private key, your message is irretrievably lost, and is unrecoverable
+Pulsar does not store the encryption key anywhere in the pulsar service. If you lose ordelete the private key, your message is irretrievably lost, and is unrecoverable.
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]:
```suggestion
Pulsar does not store the encryption key anywhere in the Pulsar service. If you lose or delete the private key, your message is irretrievably lost, and is unrecoverable.
```

Comment (Jennifer88huang) [reply site2/docs/security-encryption.md]: "Pulsar", "Apache Pulsar" are proper nouns for this project.

File: site2/docs/security-encryption.md [hunk 3]
```diff
@@ -26,17 +26,23 @@ Pulsar does not store the encryption key anywhere in the pulsar service. If you
...
+
 2. Add the public and private key to the key management and configure your producers to retrieve public keys and consumers clients to retrieve private keys.
-3. Implement CryptoKeyReader::getPublicKey() interface from producer and CryptoKeyReader::getPrivateKey() interface from consumer, which will be invoked by Pulsar client to load the key.
+
+3. Implement CryptoKeyReader::getPublicKey() interface from producer and CryptoKeyReader::getPrivateKey() interface from consumer, which Pulsar client invokes to load the key.
+
 4. Add encryption key to producer configuration: conf.addEncryptionKey("myapp.key")
-5. Add CryptoKeyReader implementation to producer/consumer config: conf.setCryptoKeyReader(keyReader)
+
+5. Add CryptoKeyReader implementation to producer or consumer config: conf.setCryptoKeyReader(keyReader)
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: do we need to add a period at the end of the sentence?

File: site2/docs/security-encryption.md [hunk 4]
```diff
@@ -141,29 +148,29 @@ pulsarClient.close();
 ```
 
 ## Key rotation
-Pulsar generates new AES data key every 4 hours or after a certain number of messages are published. The asymmetric public key is automatically fetched by producer every 4 hours by calling CryptoKeyReader::getPublicKey() to retrieve the latest version.
+Pulsar generates new AES data key every 4 hours or after publishing a certain number of messages. Producer fetch the asymmetric public key every 4 hours by calling CryptoKeyReader::getPublicKey() to retrieve the latest version.
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: Producer fetch?
Producers fetch
or
A producer fetches

File: site2/docs/security-encryption.md [hunk 5]
```diff
@@ -141,29 +148,29 @@ pulsarClient.close();
...
+
+## Enable encryption at the producer application:
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: do not add any punctuation for headings
check all similar cases.

File: site2/docs/security-encryption.md [hunk 6]
```diff
@@ -141,29 +148,29 @@ pulsarClient.close();
...
+If you produce messages that are consumed across application boundaries, you need to ensure that consumers in other applications have access to one of the private keys that can decrypt the messages. You can do this in two ways:
+1. The consumer application provides you access to their public key, which you add to your producer keys.
+2. You grant access to one of the private keys from the pairs that producer uses. 
 
-## Enabling encryption at the producer application:
-If you produce messages that are consumed across application boundaries, you need to ensure that consumers in other applications have access to one of the private keys that can decrypt the messages.  This can be done in two ways:
-1. The consumer application provides you access to their public key, which you add to your producer keys
-1. You grant access to one of the private keys from the pairs used by producer 
+In some cases, the producer may want to encrypt the messages with multiple keys. For this, add all such keys to the config. Consumer is able to decrypt the message, as long as the consumer has access to at least one of the keys.
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: Check and refine this sentence.

File: site2/docs/security-encryption.md [hunk 7]
```diff
@@ -141,29 +148,29 @@ pulsarClient.close();
...
 
-In some cases, the producer may want to encrypt the messages with multiple keys. For this, add all such keys to the config. Consumer will be able to decrypt the message, as long as it has access to at least one of the keys.
+Here is an example if you need to encrypt the messages using 2 keys (myapp.messagekey1 and myapp.messagekey2):
```
Comment (Jennifer88huang) [inline site2/docs/security-encryption.md]: Try to not use "Here is" in technical writing. Check IBM style guides.

**Comment by Monica-zy** [comment]
run java8 tests

**Comment by Monica-zy** [comment]
run java8 tests

**Comment by Monica-zy** [comment]
run java8 tests

**Pull request merged**